jobs/ssh/regenerate_puppet_certificate.erb

```diff
@@ -0,0 +1,25 @@
+<%#
+name: Regenerate puppet certificate
+job_category: Puppet
+description_format: Regenerate puppet certificate
+snippet: false
+provider_type: SSH
+kind: job_template
+%>
+
+<% # this template requires safe mode to be disabled -%>
+
+<%= render_template('Service Action - SSH Default', :service => 'puppet', :action => 'stop') -%>
+
+<% if preview? -%>
+#   this would delete a certificate on puppet ca, but running in preview mode
+<% else -%>
+  <%= @host.puppet_ca_proxy.statuses[:puppetca].destroy(@host.name) && '# deleted cert on proxy' %>
```

Comment:
This can't be done since I have no control on when the template will be executed. Note that the code evaluates on template rendering so autosign creation and deletion would be done even before smart proxy copies the script on target. Users need to setup autosign manually e.g. to * or they need to sign the request afterwards. All can be done through Foreman UI.